back-nest/src/redis/redis.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, OnModuleDestroy, OnModuleInit } from '@nestjs/common';
+import { Logger } from '@nestjs/common';
+import Redis from 'ioredis';
+
+@Injectable()
+export class RedisService implements OnModuleInit, OnModuleDestroy {
+  private readonly logger = new Logger(RedisService.name);
+  private redisClient: Redis;
+
+  async onModuleInit() {
+    this.redisClient = new Redis({
+      host: process.env.REDIS_HOST,
+      port: parseInt(process.env.REDIS_PORT, 1) || 6379,
+    });
+
+    this.redisClient.on('connect', () => this.logger.log('Connected to Redis'));
+    this.redisClient.on('error', (err) =>
+      this.logger.error('Redis error:', err),
+    );
+  }
+
+  async set(
+    key: string,
+    value: any,
+    expirationInSeconds?: number,
+  ): Promise<void> {
+    const serializedValue = JSON.stringify(value);
+    if (expirationInSeconds) {
+      await this.redisClient.set(
+        key,
+        serializedValue,
+        'EX',
+        expirationInSeconds,
+      );
+    } else {
+      await this.redisClient.set(key, serializedValue);
+    }
+  }
```

Comment:
You can simply avoid nested logic with else by putting return statement

```suggestion
    if (expirationInSeconds) {
      await this.redisClient.set(
        key,
        serializedValue,
        'EX',
        expirationInSeconds,
      );
  
      return;
    }

    await this.redisClient.set(key, serializedValue);
  }
```